src/test/scala/com/ubirch/services/IncidentListenerSpec.scala

```diff
@@ -117,9 +125,218 @@ class IncidentListenerSpec extends TestBase with EmbeddedKafka with StrictLoggin
+      val hardWareId2 = "HardwareId2"
+      val hwIdHeader1 = new RecordHeader(HeaderKeys.X_UBIRCH_HARDWARE_ID, hardWareId1.getBytes)
+      val hwIdHeader2 = new RecordHeader(HeaderKeys.X_UBIRCH_HARDWARE_ID, hardWareId2.getBytes)
+      val headers = new RecordHeaders().add(hwIdHeader1).add(hwIdHeader2)
+      val customerRecord: ConsumerRecord[String, Array[Byte]] = new ConsumerRecord("", 0, 0, 0, TimestampType.NO_TIMESTAMP_TYPE, 0L, 0, 0, "", Array.emptyByteArray, headers)
+      val result = mockIncidentListener.retrieveHeader(customerRecord, HeaderKeys.X_UBIRCH_HARDWARE_ID)
+      result mustBe Left(mockIncidentListener.HeaderError(2, HeaderKeys.X_UBIRCH_HARDWARE_ID))
+    }
+  }
+
+  "createIncidentFromCR" - {
+    "(success) create NiomonError incident" in {
+      val niomonError = NiomonError("NoSuchElementException: Header with key x-ubirch-hardware-id is missing. Cannot verify msgpack.", Seq("authentication error"), "niomon-decoder", "7c9743e7-fa43-4790-996d-59d9ee06f2a6")
+      val niomonErrorJson: String =
+        s"""{
+           |  "error": "${niomonError.error}",
+           |  "causes": ["${niomonError.causes(0)}"],
+           |  "microservice": "${niomonError.microservice}",
+           |  "requestId": "${niomonError.requestId}"
+           |}""".stripMargin
+
+      val errorCode = "errorCode"
+      val hardWareId = "HardwareId"
+      val hwIdHeader = new RecordHeader(HeaderKeys.X_UBIRCH_HARDWARE_ID, hardWareId.getBytes)
+      val xCodeHeader = new RecordHeader(HeaderKeys.X_CODE, errorCode.getBytes)
+      val niomonValue = niomonErrorJson.getBytes(StandardCharsets.UTF_8)
+      val headers = new RecordHeaders().add(xCodeHeader).add(hwIdHeader)
+      val customerRecord: ConsumerRecord[String, Array[Byte]] = new ConsumerRecord(niomonErrorTopic, 0, 0, 0, TimestampType.NO_TIMESTAMP_TYPE, 0L, 0, 0, "", niomonValue, headers)
+
+      val mockIncidentListener = getMockInjectListener()
+      val result: Either[Throwable, Incident] = mockIncidentListener.createIncidentFromCR(customerRecord, hardWareId, Some(errorCode))
+      result.isRight mustBe true
+      result.right.foreach { incident =>
+        incident.requestId mustBe niomonError.requestId
+        incident.hwDeviceId mustBe hardWareId
+        incident.errorCode mustBe Some(errorCode)
+        incident.error mustBe niomonError.error
+        incident.microservice mustBe niomonError.microservice
+      }
+    }
+
+    "(success) create EventLogError incident" in {
+      val errorDateStr = "2020-12-01T08:52:09.484Z"
+      val formatter = DateTimeFormat.forPattern("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'");
+      val errorDate = formatter.parseDateTime(errorDateStr).toDate
+      val event = Event("", "Error in the Encoding Process: No CustomerId found", "", JString(""), errorDate, "event-log-service")
+      val eventLogError = EventlogError(JObject(List.empty[JField]), "", "", "", "", event, errorDate, "", "", Seq.empty[JValue])
+
+      val eventLogErrorJson: String =
```

Comment:
Would it maybe have been more efficient to use one of the TestErrorMessages that already exists or do you think it becomes more clear orhas other benefits, to create the error in this test itself?

Reply:
one of the TestErrorMessages that already exists is defined by using just string.
Mine is defined by using some variables of `eventLogError` case class to assert each element in test.

Reply:
So, one of the TestErrorMessages is a completely raw JSON string, it's used for an interface test such as the `IncidentListenerSpec` class.
my `errorLogErrorJson` depending on some variables is used for a small unit test such as `IncidentListenerFuncSpec` class.

If we write more test code with a similar JSON string in the future, we should find an efficient way.

Reply:
sounds good!